web-app/src/components/BookingModal/container.js

```diff
@@ -163,15 +182,39 @@ const Container = Wrapped =>
         });
     };
 
+    getLegacyMessages = () => {
+      const { legacyMessages } = this.state;
+      event.preventDefault();
+      if (!legacyMessages.length) {
+        getAllHolidays()
+          .then(() => {
+            //temporarily overwriting with test data as end point data not yet available
+            this.setState({ legacyMessages: testMessageArray });
+          })
+          .catch(error => {
+            Swal({
+              title: 'Error',
+              text: error.message,
+              type: 'error',
+            });
+          });
+      }
+    };
+
     render() {
+      const { toggleRejectionMessageView, legacyMessages, toggleRejectionResponseView, loading } = this.state;
       return (
         this.props.employeeId && (
           <Wrapped
+            legacyMessages={legacyMessages}
+            legacyHolidayMessagelist={this.legacyHolidayMessagelist}
             submitRejectionResponse={this.submitRejectionResponse}
+            toggleRejectionMessageView={toggleRejectionMessageView}
             rejectionResponseText={this.state.rejectionResponseText}
             assignRejectionResponseText={this.assignRejectionResponseText}
             booking={this.props.booking}
-            toggleRejectionResponseView={this.state.toggleRejectionResponseView}
+            toggleLegacyHolidayMessageView={this.toggleLegacyHolidayMessageView}
+            toggleRejectionResponseView={toggleRejectionResponseView}
```

Comment:
Some of this stuff definitely needs to be extracted into its own component in the future. The amount of props we're passing down is getting a bit mad.